template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CentralCmTemplateUpdaterTest.java

```diff
@@ -60,14 +61,14 @@
     private RDSConfig rdsConfig;
 
     @Before
-    public void setUp() throws IOException {
+    public void setUp() {
         List<CmTemplateComponentConfigProvider> cmTemplateComponentConfigProviders = List.of(new HiveMetastoreConfigProvider());
         when(cmTemplateProcessorFactory.get(anyString())).thenAnswer(i -> new CmTemplateProcessor(i.getArgument(0)));
         when(templatePreparationObject.getClusterDefinitionView()).thenReturn(clusterDefinitionView);
         when(templatePreparationObject.getGeneralClusterConfigs()).thenReturn(generalClusterConfigs);
         when(templatePreparationObject.getRdsConfigs()).thenReturn(getRdsConfigs());
         when(generalClusterConfigs.getClusterName()).thenReturn("testcluster");
-        when(cmTemplateComponentConfigProcessor.getCmTemplateComponentConfigProviderList()).thenReturn(cmTemplateComponentConfigProviders);
+        ReflectionTestUtils.setField(cmTemplateComponentConfigProcessor, "cmTemplateComponentConfigProviderList", cmTemplateComponentConfigProviders);
```

Comment:
what is the problem with _when()_?

Reply:
We used an extra method just to mock with `when` i removed the additional method, so we can't use `when` any more